data-types.md

```diff
@@ -3,4 +3,408 @@ title: Data Types
+|           a            |         b          |
++------------------------+--------------------+
+| 1.01234567890123456789 | 2.0123456789012346 |
++------------------------+--------------------+
+# The value in "a" is precise, while the value in "b" has been limited to 17 digits.
+~~~
+
+## `FLOAT`
+
+The `FLOAT` type stores inexact, floating-point numbers with up to 17 digits in total and at least one digit to the right of the decimal point. 
+
+#### Synonyms
+
+In CockroachDB, the following are synonyms of `FLOAT` and are implemented identically:
+
+- `REAL` 
+- `DOUBLE PRECISION` 
+
+#### Format
+
+When declaring a float, format it as `1.2345`. 
+
+Alternately, you can declare a float as `+Inf` (positive infinity), `-Inf` (negative infinity), or `NaN` (not a number) as follows:
+
+- `CAST('+Inf' AS FLOAT)`
+- `CAST('-Inf' AS FLOAT)`
+- `CAST('NaN' AS FLOAT)`
+
+#### Examples
+
+~~~
+CREATE TABLE floats (a FLOAT PRIMARY KEY, b REAL, c DOUBLE PRECISION);
+
+SHOW COLUMNS FROM floats;
++-------+-------+-------+---------+
+| Field | Type  | Null  | Default |
++-------+-------+-------+---------+
+| a     | FLOAT | false | NULL    |
+| b     | FLOAT | true  | NULL    |
+| C     | FLOAT | true  | NULL    |
++-------+-------+-------+---------+
+
+INSERT INTO floats VALUES (1.012345678901, 2.01234567890123456789, CAST('+Inf' AS FLOAT));
+
+SELECT * FROM floats;
++----------------+--------------------+------+
+|       a        |         b          |  c   |
++----------------+--------------------+------+
+| 1.012345678901 | 2.0123456789012346 | +Inf |
++----------------+--------------------+------+
+# Note that the value in "a" has been limited to 17 digits.
+~~~
+
+## `BOOL`
+
+The `BOOL` type stores a Boolean value of `false` or `true`. 
+
+#### Synonyms
+
+In CockroachDB, `BOOLEAN` is a synonym of `BOOL` and is implemented identically. 
+
+#### Format
+
+When declaring a `BOOL`, format it as `false` or `true`. 
+
+Alternately, you can cast `0` or `1` as a `BOOL`:
+
+- `0::BOOL` (false) 
+- `CAST (0 AS BOOL)` (false)
+- `1::BOOL` (true)
+- `CAST (1 AS BOOL)` (true)
+
+#### Examples
+
+~~~
+CREATE TABLE bool (a INT PRIMARY KEY, b BOOL, c BOOLEAN);
+
+SHOW COLUMNS FROM bool;
++-------+------+-------+---------+
+| Field | Type | Null  | Default |
++-------+------+-------+---------+
+| a     | INT  | false | NULL    |
+| b     | BOOL | true  | NULL    |
+| c     | BOOL | true  | NULL    |
++-------+------+-------+---------+
+
+INSERT INTO bool VALUES (12345, true, 0::bool);
+
+SELECT * FROM bool;
++-------+------+-------+
+|   a   |  b   |   c   |
++-------+------+-------+
+| 12345 | true | false |
++-------+------+-------+
+~~~
+
+## `DATE`
+
+The `DATE` type stores a date.
+
+#### Format
+
+When declaring a `DATE`, format it as `DATE '2016-01-25'`. 
+
+Alternately, you can format a date as follows:
+
+- `'2016-01-25'::DATE`
+- `CAST ('2016-01-25' AS DATE)` 
+
+Note that a date is stored with hour, minute, second, and timezone set to 0.
```

Comment:
Dates don't really have these fields, so it's confusing to talk about them being stored that way. Rather than talk about how things are stored, say something like "in some contexts, dates may be displayed with hours, minutes, seconds and timezone set to zero". (it's kind of a bug whenever this happens, but it's harder to fix than it looks)